gradio/interface.py

```diff
@@ -166,12 +166,12 @@ def __init__(
         )
 
         self.interface_type = self.InterfaceTypes.STANDARD
-        if inputs is None and outputs is None:
+        if (inputs is None or inputs == []) and (outputs is None or outputs == []):
```

Comment:
since you call super.init() first, you could just check if self.inputs == [].

Reply:
Where in `Blocks()` do I assign `self.inputs`?